.github/workflows/draft-release.yml

```diff
@@ -0,0 +1,19 @@
+name: Draft Release
+on:
+  schedule:
+    - cron: '0 1 1 * *'
+    # At 01:00 on the first day of every month
+  workflow_dispatch:
```

Comment:
Having written this and thought about how we're going to actually use it, there's a few related issues giving me pause for thought here which are sort of tied together (at least in my mind).

* I think it is useful for the release tag to include the full date.
* We're mostly going to do one release a month. Occasionally (I'm thinking of the situation where there is a security fix) it will be useful to do an out-of-sequence release.
* We generate the title for the changelog entry using the current date when we generate the PR here https://github.com/badges/shields/pull/6102/files#diff-06ec58a42aa9b0bdf423c39dc6b326395f69db72e0127e3e1d3f83b9c87d27f4R18 and then we merge it, we generate the tag name again using the current date here https://github.com/badges/shields/blob/2b3ad63748bf643d61d16721892c9f1795ce0ab6/.github/workflows/tag-release.yml#L20 . If we review/merge the PR the day it is submitted, this is fine, but:
  * If we leave it until another day the title we generated on the changelog entry is now different than the tag name we will generate when we merge it. We need to manually edit it.
  * In fact, if we merge any other PRs after the changelog PR is opened then the suggested changelog entry we generated is now out of date. We may need to manually edit it.

I guess what we ideally want is like a `dependabot rebase` command but in lieu of that I think the solution I've arrived at is this.. the action runs once a month automatically on a schedule, but it can also be triggered manually if we want (workflow_dispatch basically gives us a 'run workflow' button).

![Screenshot at 2021-01-24 19-27-22](https://user-images.githubusercontent.com/6025893/105643280-d2313180-5e86-11eb-99de-80d4167949a0.png)

Then that gives us a reminder every month, but if we ignore it for a bit and need to re-generate the PR (or run the same process mid-month) we can do. Also because we generate a unique branch name we can run the workflow multiple times on the same day if needed. I think that gives us a reasonable balance of those constraints. Anyone else got thoughts on the trade-offs here? Any ideas on how we could deal with it better or make this more fool-proof?

Reply:
That's a good observation. My initial reaction to the above comment https://github.com/badges/shields/pull/6102/files#r563348469 was that it sounded perfectly reasonable, though I guess that is part of a minor wrinkle here in that the odds of the log entries vs commits getting out of sync (seems like a much lower chance that there'd be self-hosted labeled PR in between).

I almost wonder if it might be easier to remove the PR and potential human delay from this, and just let the robots do their thing fully automated (permit auto merging/squashing or even push directly). We'd always have the ability to edit the changelog after, and even in the the case mentioned in #5866 about a big pending self-hosted change landing shortly after, we could always kick off manually.

> Any ideas on how we could deal with it better or make this more fool-proof?

I guess another thing we could do is add another action that runs daily, checks for these release PRs, and squash merges if it finds one open and approved, or if there's one open but not approved then closes the PR and re-runs the workflow so that a new one is generated with the latest?

That obviously won't solve any deltas in intra-day commits, but could potentially help prevent the tag and entry from getting out of sync

Reply:
I wouldn't be totally against just letting the bots do their thing (I for one welcome our new release-bot overlords).

One thing I don't really have a handle on yet is how many non-dependabot commits we generate in a typical month and how many of them are just completely irrelevant to self-hosting users (i.e: what's the signal-to-noise ratio of just dumping this to the log with no manual review). I suspect we won't figure this out until we try it for a bit, or maybe I should go and do some research. We have plenty of history to dive into.

While it is true that we can always edit the changelog later, I think as far as possible it is useful if the changelog entry in the tagged ref is final if possible. i.e: if I browse the repo at https://github.com/badges/shields/tree/server-2021-01-18 it would be nice if the changelog file has a useful entry at that commit, rather than being "raw" there and then tidied in a later commit.

Reply:
> (I for one welcome our new release-bot overlords).

:laughing: I keep catching myself trying to use them in non-applicable places.. like asking Dependabot to rebase a humans PR or adding a squash label in a project that doesn't have Ranger

> I suspect we won't figure this out until we try it for a bit, or maybe I should go and do some research. We have plenty of history to dive into

Agreed. I've a slight lean towards the bot overlords but no strong opinions. Think it'd be best to just give this a try and learn/adjust as needed